src/components/Avatar.js

```diff
@@ -44,16 +53,25 @@ class Avatar extends PureComponent {
             this.props.size === 'small' ? styles.avatarSmall : styles.avatarNormal,
             ...this.props.imageStyles,
         ];
+        
+
+           
         return (
             <View pointerEvents="none" style={this.props.containerStyles}>
                 {this.props.isDefaultChatRoom
                     ? <RoomAvatar avatarStyle={imageStyle} isArchived={this.props.isArchivedRoom} />
-                    : <Image source={{uri: this.props.source}} style={imageStyle} />}
+                    :  this.props.network.isOffline ? <Image source={require('../../assets/images/default_avatar.png')} style={imageStyle}/> : <Image source={{uri: this.props.source}} style={imageStyle}/>
+                                      
+                    }
```

Comment:
A few things here...
1) I requested that the `require` statement not be used inline on the original issue.
2) The use of nested ternaries here is not acceptable.

Please make the `isOffline` check outside of the return statement. Give this class a constructor with a member variable `this.avatarSource`, and set that based on the `isOffline` prop.